### PR TITLE
Add net.sourceforge.symphonic.jsymphonic

### PR DESCRIPTION
**Pull request by georgewoodall82**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  A file manager for old sony network walkman devices

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** unable to contact upstream as the program is quite old, would i need to make a fork?

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: flathub.json
Comment (bbhtt) [inline flathub.json:1]: Why is this x86_64 only?

File: net.sourceforge.symphonic.jsymphonic.desktop
Comment (bbhtt) [inline net.sourceforge.symphonic.jsymphonic.desktop:1]: This should be in the upstream repo

File: net.sourceforge.symphonic.jsymphonic.metainfo.xml
Comment (bbhtt) [inline net.sourceforge.symphonic.jsymphonic.metainfo.xml:1]: This should be in the upstream repo

File: net.sourceforge.symphonic.jsymphonic.png
Comment (bbhtt) [inline net.sourceforge.symphonic.jsymphonic.png:1]: This should be in the upstream repo

File: net.sourceforge.symphonic.jsymphonic.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+id: net.sourceforge.symphonic.jsymphonic
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --filesystem=host
```
Comment (bbhtt) [inline net.sourceforge.symphonic.jsymphonic.yml:7]: Why is host needed?

File: net.sourceforge.symphonic.jsymphonic.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - install -D ${FLATPAK_ID}.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -D ${FLATPAK_ID}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm755 jsymphonic.sh /app/bin/jsymphonic
+      - install -D *.jar /app/jsymphonic/jsymphonic.jar
+    sources:
+      - type: script
+        dest-filename: jsymphonic.sh
+        commands:
+          - mkdir -p /var/config/jsymphonic
+          - cd /var/config/jsymphonic/
```
Comment (bbhtt) [inline net.sourceforge.symphonic.jsymphonic.yml:27]: Why is this required? Why can't the app initiate its own config directory?

**Comment by bbhtt** [comment]
The upstream status says abandonded https://sourceforge.net/projects/symphonic/ ?

File: flathub.json
Comment (georgewoodall82) [reply flathub.json:1]: i just removed the json file, it should build for arm64 too now

File: net.sourceforge.symphonic.jsymphonic.desktop
Comment (georgewoodall82) [reply net.sourceforge.symphonic.jsymphonic.desktop:1]: I have no access to the upstream repo as it is abandoned

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 1 above)
Comment (georgewoodall82) [reply net.sourceforge.symphonic.jsymphonic.yml:7]: it needs to access usb for the sony network walkman, and the home folder for music files to import and export. it doesn't support portals.

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 2 above)
Comment (georgewoodall82) [reply net.sourceforge.symphonic.jsymphonic.yml:27]: it creates the config in the same directory that it is run, so the config will be different for every place you run it by default. this makes it the same regardless of where it is ran.

**Comment by georgewoodall82** [comment]
> The upstream status says abandonded https://sourceforge.net/projects/symphonic/ ?

Yeah it is, do i need to specify that in the metainfo.xml?

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 1 above)
Comment (hfiguiere) [reply net.sourceforge.symphonic.jsymphonic.yml:7]: `host` doesn't give access to USB.

Ad for music. `xdg-music`

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 2 above)
Comment (hfiguiere) [reply net.sourceforge.symphonic.jsymphonic.yml:27]: this is not persistent in any way.

File: net.sourceforge.symphonic.jsymphonic.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          - /app/jre/bin/java -jar /app/jsymphonic/jsymphonic.jar
+      - type: archive
+        url: https://downloads.sourceforge.net/project/symphonic/jsymphonic/JSymphonic version 0.3.0 Ode To Freedom/JSymphonic_v0.3.0_Ode_To_Freedom.zip
```
Comment (hfiguiere) [inline net.sourceforge.symphonic.jsymphonic.yml:30]: what is this and why is it needed?

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 1 above)
Comment (georgewoodall82) [reply net.sourceforge.symphonic.jsymphonic.yml:7]: Usbs are usually mounted at /run/media or /media which host gives access to

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 2 above)
Comment (georgewoodall82) [reply net.sourceforge.symphonic.jsymphonic.yml:27]: Would it be better to leave it as the default behaviour? (Creating the config in the current directory)

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 1 above)
Comment (hfiguiere) [reply net.sourceforge.symphonic.jsymphonic.yml:7]: That's not USB. That's removable drives.

File: net.sourceforge.symphonic.jsymphonic.yml (hunk 2 above)
Comment (hfiguiere) [reply net.sourceforge.symphonic.jsymphonic.yml:27]: no. Fixing the code is the best.

**Comment by bbhtt** [comment]
> Yeah it is, do i need to specify that in the metainfo.xml?

No that's fine. Are you willing to maintain this/patch things, if necessary to keep the Flatpak updated and working? Every year or so you have to bump the runtime to a major version as old ones are EOL-ed.

Upstream being gone means you likely need to patch it in times like that.

**Comment by georgewoodall82** [comment]
just made a fork that may fix these issues, ill make a new pull request
